src/main/java/ch/ivyteam/ivy/maven/util/MavenDependencies.java

```diff
@@ -0,0 +1,109 @@
+package ch.ivyteam.ivy.maven.util;
+
+import java.io.File;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.project.MavenProject;
+
+/**
+ * @since 9.2.2
+ */
+public class MavenDependencies
+{
+  private final MavenProject project;
+  private final MavenSession session;
+  private String typeFilter;
+
+  public MavenDependencies(MavenProject project, MavenSession session)
+  {
+    this.project = project;
+    this.session = session;
+  }
+  
+  public MavenDependencies type(String newTypeFilter)
+  {
+    this.typeFilter = newTypeFilter;
+    return this;
+  }
+  
+  public List<File> localTransient()
+  {
+    Stream<Artifact> artifacts = stream(project.getArtifacts())
+      .filter(this::isLocalDep);
+    return getFiles(artifacts);
+  }
+  
+  private boolean isLocalDep(Artifact artifact)
+  {
+    return artifact.getDependencyTrail().stream()
```

Comment:
ah is this the way to get the full dependecy stack?

Reply:
yes: it will print the full chain of parents that lead to this dependency ... so if we have multiple 'iar' parents ... the dependency can be ignored.